docs/docs/building-blocks/6-optimizers.md

```diff
@@ -38,36 +38,32 @@ All of these can be accessed via `from dspy.teleprompt import *`.
 
-4. **`BootstrapFewShotWithOptuna`**: Applies `BootstrapFewShot` through Optuna hyperparameter optimization across demonstration sets, running trials to maximize evaluation metrics. As before, chooses the best of a candidate set of demonstration examples, but this time using Optuna to make the choice. <!--- TBQH, I don't understand how Optuna does this. As far as I can tell it simply chooses best based on multiple evaluations, rather than a single one, and mention of "hyperparameters" seems to be a red herring. ---> 
+4. **`BootstrapFewShotWithOptuna`**: Applies `BootstrapFewShot` through Optuna hyperparameter optimization across demonstration sets, running trials to maximize evaluation metrics and selecting the best demonstrations. 
 
-5. **`KNNFewShot`**. Selects demonstrations through k-Nearest Neighbors algorithm.  Vectorizes the examples, and then clusters them, using cluster centers with `BootstrapFewShot` for bootstrapping/selection process. <!-- I'm not at all sure that this is right. I couldn't follow the KNN code in the repo, so I just assumed that dspy was trying to cover the space of possible examples by picking centers of different clusters. -->
+5. **`KNNFewShot`**. Selects demonstrations through k-Nearest Neighbors algorithm.  Vectorizes the examples, and then clusters them, using cluster centers with `BootstrapFewShot` for bootstrapping/selection process. 
 
 
 
 #### Automatic Instruction Optimization
 
-Unlike the above optimizers, these attempt to optimize the instructions, rather than the questions.
+These optimizers serve to produce optimal instructions for the prompt, in addition to optimized few-shot demonstrations.
 
-6. **`COPRO`**: Uses an LLM to generate and refine new instructions for each step, and optimizes them with coordinate ascent (hill-climbing using the metric function and the `trainset`).  The number of iterations of prompt improvement is controlled by the `depth` parameter.
-
-#### Simultaneous optimization
-
-This optimizer combines automatic few-shot learning and instruction optimization in one process.
+6. **`COPRO`**: Generates and refines new instructions for each step, and optimizes them with coordinate ascent (hill-climbing using the metric function and the `trainset`). Parameters include `depth` which is the number of iterations of prompt improvement the optimizer runs over.
 
 7. **`MIPRO`**: Generates instructions *and* few-shot examples in each step. The instruction generation is data-aware and demonstration-aware. Uses Bayesian Optimization to effectively search over the space of generation instructions/demonstrations across your modules.
 
 
 #### Automatic Finetuning
 
-This optimizer is used to fine-tune the underlying LLM(s) instead of the prompting.
```

Comment:
I think it would help to distinguish this from the other optimizers.  Maybe add "but not a few-shot example set, or the instructions."?

Reply:
similar comment as above with document sections